### ISSUE TITLE
[IMPROVEMENT]: Formation - CONSIDER making Featured Content heading dynamic

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/vets-design-system-documentation#244]
@1Copenut commented on [Thu Apr 23 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8352)

**Feedback framework**

- **✔️ Consider** for suggestions/enhancements 

## Description

The Featured Content component doesn't appear to have a React version, and would benefit from a way to make the hard-coded `<h3>` dynamic, so developers could control what heading level is output to the page. Screenshot attached below.

## Related Issues
* [Issue request to create this component](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/255)

## Point of Contact

**VFS Point of Contact:** _Trevor or Travis_

## Acceptance Criteria

- [ ] Confirm we have a React component for Featured Content
- [ ] Provide a way (prop?) to dynamically control heading level
- [ ] Ensure good test coverage for this new functionality
- [ ] Create and/or modify unit tests for this component to include an aXe scan

## Environment

* https://design.va.gov/components/featured-content

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

![Screen Shot 2020-04-23 at 3 45 45 PM](https://user-images.githubusercontent.com/934879/80148536-ccff2900-857a-11ea-8ced-2a5d6b7e99d0.png)

**Comment by holdenhinkle** [comment]
@pjhill I can't find any header components (`h1`-`h6`). I can create one, but I'm curious, why only `h3`? I quickly reviewed a number of `vets-website` apps, and I don't think anyone is missing this.

**Comment by holdenhinkle** [comment]
PR - https://github.com/department-of-veterans-affairs/component-library/pull/164

**Comment by holdenhinkle** [comment]
Hi @GnatalieH (cc @pjhill )

I'm a bit unclear about what to do for this ticket.

First of all, a Featured Content React component does not exist. Should I create one?

This ticket asks for a Featured Content Heading component, which I created ([see the pr](https://github.com/department-of-veterans-affairs/component-library/pull/164)). However, I don't think this component is very usable, and I think most engineers would rather write the `html` instead of using this component.

Consider using the component...
```js
<FeaturedContentHeading
  level={3}
  text="Here's the heading text"
/>
```

... vs just writing HTML
```html
<h3>Here's the heading text</h3>
```

Instead of just making a 'Featured Content Heading' component, I could create a 'Featured Content' component which would be basically the same thing as the 'Featured Content Heading' component but additional content under the heading could be rendered using `props.children`:

```js
<FeaturedContent level={3}  text="Here's the heading text">
  <p>Any html content can go here, like this:</p>
  <ul>
    <li>First bullet</li>
  </ul>
</FeaturedContent>
```

Anyway, please let me know what you think.

Thanks!

**Comment by holdenhinkle** [comment]
PR for new FeaturedContent component - https://github.com/department-of-veterans-affairs/component-library/pull/164

**Comment by GnatalieH** [comment]
@holdenhinkle is [this Storybook](https://design.va.gov/storybook/?path=/docs/components-featured-content--default-story) not a React component?

**Comment by holdenhinkle** [comment]
That's not a React component.

**Comment by holdenhinkle** [comment]
It's exactly what you see when you click 'Show Code':

```
<div class="feature">
      <h3>If I’m a Veteran, can I get VR&amp;E benefits and services?</h3>
      <p>You may be eligible for VR&amp;E benefits and services if you’re a Veteran, and you meet all of the requirements listed below.</p>
      <p><strong>All of these must be true. You:</strong></p>
      <ul>
        <li>Didn’t receive a dishonorable discharge, <strong>and</strong></li>
        <li>Have a service-connected disability rating of at least 10% from VA, <strong>and</strong></li>
        <li><a href="#">Apply for VR&amp;E services</a></li>
      </ul>
    </div>
```

**Comment by holdenhinkle** [comment]
Sorry for another post :-)

Here's the link to the story that page is generated from - https://github.com/department-of-veterans-affairs/component-library/blob/master/stories/featured-content.stories.mdx

**Comment by GnatalieH** [comment]
Thanks @holdenhinkle. I'll defer to @bkjohnson and @cvalarida on this since they are the design system team developers.

**Comment by bkjohnson** [comment]
We are migrating our existing components away from React and into Web Components. One of the reasons for this is so that they can be used outside of React applications. As a result, we aren't looking to add any _new_ React components only to have to convert them down the road.

**Comment by holdenhinkle** [comment]
PR - https://github.com/department-of-veterans-affairs/component-library/pull/173
